Roles/Impostor/DollMaster.cs

```diff
@@ -0,0 +1,335 @@
+        }
+        else if (controllingTarget != null)
+        {
+            UnPossess(pc, Utils.GetPlayerById(controllingTarget.PlayerId));
+            GetPlayersPositions(pc);
+            SwapPlayersPositions(pc);
+        }
+        return false;
+    }
+
+    // Possess Player
+    private static void Possess(PlayerControl pc, PlayerControl target, bool shouldAnimate = false)
+    {
+        (target.MyPhysics.FlipX, pc.MyPhysics.FlipX) = (pc.MyPhysics.FlipX, target.MyPhysics.FlipX); // Copy the players directions that they are facing, Note this only works for modded clients!
+        pc.RpcShapeshift(target, shouldAnimate);
+        target.RpcShapeshift(pc, shouldAnimate);
+        pc.Notify(Utils.ColorString(Utils.GetRoleColor(CustomRoles.DollMaster), GetString("DollMaster_PossessedTarget")));
+    }
+
+    // UnPossess Player
+    private static void UnPossess(PlayerControl pc, PlayerControl target, bool shouldAnimate = false)
+    {
+        WaitToUnPossess = false;
+        (target.MyPhysics.FlipX, pc.MyPhysics.FlipX) = (pc.MyPhysics.FlipX, target.MyPhysics.FlipX); // Copy the players directions that they are facing, Note this only works for modded clients!
+        pc.RpcShapeshift(pc, shouldAnimate);
+        target.RpcShapeshift(target, shouldAnimate);
+        pc.RpcResetAbilityCooldown();
+
+        IsControllingPlayer = false;
+        ResetPlayerSpeed = true;
+        _ = new LateTask(() =>
+        {
+            ReducedVisionPlayers.Clear();
+            if (TargetDiesAfterPossession.GetBool() && !GameStates.IsMeeting) target.RpcMurderPlayer(target);
+        }, 0.35f);
+    }
+
+    // Get players locations.
+    private static void GetPlayersPositions(PlayerControl pc)
+    {
+        if (controllingTarget == null) return;
+        controllingTargetPos = controllingTarget.GetCustomPosition();
+        DollMasterPos = pc.GetCustomPosition();
+    }
+
+    // Swap location with DollMaster and Target.
+    private static void SwapPlayersPositions(PlayerControl pc)
+    {
+        if (controllingTarget == null) return;
+        controllingTarget.RpcTeleport(DollMasterPos);
+        pc.RpcTeleport(controllingTargetPos);
+    }
+
+    // Set name Suffix for Doll and Main Body under name.
+    public override string GetSuffix(PlayerControl seer, PlayerControl target = null, bool isForMeeting = false)
+    {
+        if (!IsControllingPlayer) return string.Empty;
+        if (GameStates.IsMeeting) return string.Empty;
+        if (target == null) return string.Empty;
+        if (controllingTarget == null) return string.Empty;
+        if (DollMasterTarget == null) return string.Empty;
+        if (seer.PlayerId != target.PlayerId && target.PlayerId == controllingTarget.PlayerId) return "<color=#ffea00>" + GetString("DollMaster_MainBody");
+        if (seer.PlayerId == target.PlayerId && target.PlayerId == DollMasterTarget.PlayerId) return "<color=#ffea00>" + GetString("DollMaster_Doll");
+        return string.Empty;
+    }

```

Comment:
better to use GetLowerTextand also in `"<color=#ffea00>" + GetString("DollMaster_MainBody");` and `"<color=#ffea00>" + GetString("DollMaster_Doll");`, remember to end it with a `+ "</color>"`